component/http/route.go

```diff
@@ -58,6 +60,12 @@ func (rb *RouteBuilder) WithTrace() *RouteBuilder {
 	return rb
 }
 
+// WithRateLimiting enables route rate limiting.
+func (rb *RouteBuilder) WithRateLimiting(limit float64, burst int) *RouteBuilder {
+	rb.rateLimiter = rate.NewLimiter(rate.Limit(limit), burst)
```

Comment:
Small note: as per the documentation, *burst is the maximum number of tokens that can be consumed in a single call to Allow, Reserve, or Wait*. 

Since we're not actually utilizing `AllowN`, `ReserveN` or `WaitN` in our code, but only use the `Allow()` method to consume a single token, I feel it makes sense to only let the users define the `limit float64` value and leave set the `burst` value to 1.

Reply:
makes sense. should we actually use AllowN then, and give users the choice to specify burst. I kind of think burst >1 would be needed in most of the cases. Thoughts?

Reply:
From what I [read](https://godoc.org/golang.org/x/time/rate) it seems that AllowN reports *whether n events may happen at a specific time*, and the Allow method is short for `AllowN(time.Now(), 1)`.

I may be missing something, but I'm not sure why we want the user to reserve multiple (N>1) tokens in a single request, and set burst > 1?

If we set an adequate *L=limit* value, then isn't it the same to have L users performing one request or, one user performing L requests with one token each?

Reply:
I actually was drawing parallel with limit & burst setup in nginx with allow & burst here. More on that here - https://www.nginx.com/blog/rate-limiting-nginx/#:~:text=The%20burst%20parameter%20defines%20how,%2C%20or%201%20every%20100ms). Not sure if I am right though.

Basically if set rate at 10r/s -> 1 request per 100 ms, if second request comes at before 200ms  - without burst we would respond to it with 429. Instead we can set a buffer of tokens with burst, which essentially allow a burst to x requests while maintaining overall rate limit per second. 

To demo what I meant by this - you could try running a **ab** test  with burst set to 1 and to 50.

* with burst set to 50 -

`
.WithRateLimiting(50, 50))
`

below ab test runs with all requests returning 200s

`
ab -n 50  http://localhost:50000/
`
* with burst set to 1 - 

`
.WithRateLimiting(50, 1)
`
below ab test runs with 49 requests returning 429 code.

Reply:
Oh, I see! I misread the documentation, running some tests cleared that up for me.

Disregard this comment, let's just add some docs and we're in a good place!